Google.PowerShell/BigQuery/GcbqDataset.cs

```diff
@@ -20,8 +21,8 @@ namespace Google.PowerShell.BigQuery
     /// READER dataset role. The -All flag will include hidden datasets in the search results. The -Filter 
     /// flag allows you to filter results by label. The syntax to filter is "labels.name[:value]". Multiple filters 
     /// can be ANDed together by connecting with a space. See the link below for more information.
-    /// If no Project is specified, the default project will be used. This cmdlet returns a DatasetList if 
-    /// no DatasetId was specified, and a Dataset otherwise.
+    /// If no Project is specified, the default project will be used. This cmdlet returns any number of 
+    /// DatasetList.DatasetData objects if no DatasetId was specified, and a Dataset otherwise.
```

Comment:
So this is an issue that I found when I was looking into this.  Apparently a DatasetList.DatasetData is not a Dataset, nor does it contain a Dataset.  It's a subset of the information returned by the Get- command, which is an actual Dataset object.  The thing is, when you're interacting with them in powershell, they have the same fields that a Dataset does, so it should be functionally the same for everything except a typeof() check.  

One possible solution that I thought of was to use the fields from the DatasetData and create a somewhat incomplete Dataset and then return that, but that didn't seem like very elegant code.  Plus that'd end up taking a while if there were a large amount of projects.  

That's why I elected to just let the user know what the api returned, and thus what the cmdlet will return.  The main use case here anyway would be to get the DatasetReference.DatasetId and then pipe it back into a get request, or to just observe the string and send it manually in a non-scripted environment.  

What are your thoughts on this?